pkg/activator/throttler.go

```diff
@@ -132,14 +146,37 @@ func (t *Throttler) Try(rev RevisionID, function func()) error {
 	return nil
 }
 
+func (t *Throttler) activatorEndpointsUpdated(newObj interface{}) {
+	endpoints := newObj.(*corev1.Endpoints)
+	count := resources.ReadyAddressCount(endpoints)
+	atomic.StoreInt32(&t.numActivators, int32(count))
+	t.updateAllBreakerCapacity()
+}
+
+func (t *Throttler) activatorCount() int {
+	activatorCount := int(atomic.LoadInt32(&t.numActivators))
+	return minOneOrValue(activatorCount)
+}
+
+// minOneOrValue function returns num if its greater than 1
+// else the function returns 1
+func minOneOrValue(num int) int {
+	if num > 1 {
+		return num
+	}
+	return 1
+}
+
 // This method updates Breaker's concurrency.
 func (t *Throttler) updateCapacity(breaker *queue.Breaker, cc, size int) (err error) {
 	targetCapacity := cc * size
+
 	if size > 0 && (cc == 0 || targetCapacity > t.breakerParams.MaxConcurrency) {
 		// The concurrency is unlimited, thus hand out as many tokens as we can in this breaker.
 		targetCapacity = t.breakerParams.MaxConcurrency
+	} else if targetCapacity > 0 {
+		targetCapacity = minOneOrValue(targetCapacity / t.activatorCount())
```

Comment:
I think `t.activatorCount()` can be passed into the function. Basically it's constant at that point so there's no read to to atomic read each time.

Reply:
Thanks for the catch. We have updated the PR to remove this logic.

Reply:
I don't think that is correct. Since the `t.activatorCount` is now unguarded and activator might scale again.
I would suggest
changing it differently:

1. replace the interface for update/force update to take one more parameter, that is |activator-service|
1. in the update all assign it to a local variable and pass into the  each forceUpdate
1. update the other call sites.

you're holding lock over breakers, so this should be safe.

Reply:
@vagababov : Updated the PR with requested changes